HW03_Graphs.Rmd

```diff
@@ -0,0 +1,174 @@
+  geom_point(color = "white", size = 3) +
+  scale_fill_manual(values = colorg) + #Not color, but fill, serves the same purpose though.
+  scale_shape_manual(values = shapeg) +
+  labs(title = "Enzymatic activity assays",
+       x = "[Substrate] (mM)",
+       y = expression(paste("Specific activity(\u03bcmol min "^{-1}, " mg "^{-1},")"))) + #\u03bc is the greek letter micro in Unicode
+  coord_cartesian(xlim = c(0, 3.2), ylim = c(0, 4000), expand = FALSE) +
+  scale_x_continuous(breaks = seq(0, 3, 0.5)) +
+  scale_y_continuous(breaks = seq(0, 4000, 1000))
+```
+
+I left out the Michaelis-Menten fit because doing it in R is way too complex for me to program right now, but there are ways to include it with `geom_smooth()`. 
+
+## readr package
+
+For this section I simply included a *csv* file with differential scanning calorimetry (*DSC*) data.
+
+#### DSC
+
+DSC thermograms can be used to determine the melting temperature of a protein (seen at the top of each transition) and to extract some calorimetric parameters of protein denaturation. You can see the original graph here:
+
+![DSC thermograms](images/thermo.JPG)
+
+And this is the recreation of it in R:
+
+
+```{r DSC, echo = TRUE}
+
+names <- c("temp", "heat_cap", "Species")
+dsc_data <- read_csv("data/calorimetry.csv", skip = 1, col_names = names) #It already has the column names in the first row, but I wanted to play a little bit
+
+ggplot(data = dsc_data, mapping = aes(x = temp, y = heat_cap, color = Species)) +
+  geom_line(size = 2) +
+  scale_color_manual(values = colorg) + 
+  labs(title = "DSC thermograms",
+       x = "Temperature (K)",
+       y = expression(paste("Heat capacity (kJ mol "^{-1}, " K "^{-1}, ")"))) +
+  scale_x_continuous(sec.axis = sec_axis(~.-273.15, name = "Temperature (°C)", breaks = seq(70, 115, 5))) + #A good example of using secondary axis to improve the interpretation of a graph
+  scale_y_continuous(breaks = seq(-150, 200, 50)) +
+  coord_cartesian(xlim = c(340, 390), ylim = c(-170, 220), expand = FALSE)
+
+
+
+
+```
+
+
+
+## readxl package
+
+Finally, I wanted to use this package to select a single sheet from and *xlsx* file that contained the data from all the previous graphs:
+
+```{r}
+print(excel_sheets("data/thermophiles.xlsx"))
+```
+
+#### Chemical denaturation
+
+Ok, for this last one I took the liberty of changing the concept. According to the Best Practices chapter in DataCamp, secondary axes aren't always a good thing, and I agree. That's why I decided to use facets to improve this graph on chemical denaturation of one of the enzymes:
+
+![Chemical denaturation curves](images/chem_denat.JPG)
+
+It was also a good idea to practice working with facets as it was part of the Data visualization course:
+
+```{r Chemical Denaturation, echo = TRUE}
+
+#Only import the sheet called "Denaturant"
+den_data <- read_excel("data/thermophiles.xlsx", sheet = "Denaturant")
```

Comment:
I didn't know you could do this either - obviously I didn't go through all of the DataCamp videos, but looks like I'm learning enough from you!